doc/architecture/uml/rest_service_get_request.wsd

```diff
@@ -0,0 +1,133 @@
+@startuml Get-Request
+participant Plugin
+participant RestService
+participant AsyncHttpClient 
+participant "RestService::m_responseQueue" as responseQ
+participant TcpClient
+entity Server
+
+autoactivate on
+
+-> RestService: start()
+    RestService -> AsyncHttpClient: regOnResponse/Error/Closed
+    RestService <-- AsyncHttpClient
+<-- RestService
+
+note over Plugin,Server
+    Get-Request:
+end note
+
+Plugin <- : processTask runs
+    Plugin -> RestService: get(URL, PreProcessCallback)
+        RestService -> RestService: put request in m_requestQueue
+        return
+    Plugin <-- RestService: restId
+Plugin -->
+
+RestService <- : loopTask runs
+    alt (RestCmd available) && (false == m_isWaitingForResponse)
+        RestService -> RestService: m_isWaitingForResponse = true
+        return
+
+        RestService -> RestService: set m_activePreProcessCallback/m_activeRestId
+        return 
+
+        RestService -> AsyncHttpClient: begin(URL)
+        RestService <-- AsyncHttpClient 
+
+        RestService -> AsyncHttpClient: GET(restID)
+            AsyncHttpClient -> AsyncHttpClient: put Cmd with restID in m_cmdQueue
+            return
+        RestService <-- AsyncHttpClient
+    end
+RestService -->
+
+AsyncHttpClient <- : AsyncHttpClientTask runs
+    alt cmd in m_cmdQueue available
+        AsyncHttpClient -> AsyncHttpClient: takeGlobalMutex()
+        return
+
+        AsyncHttpClient -> AsyncHttpClient: getRequest()
+            AsyncHttpClient -> AsyncHttpClient: connect()
+                AsyncHttpClient -> TcpClient: connect()
+                return
+            return
+        AsyncHttpClient <-- AsyncHttpClient
+    end   
+AsyncHttpClient -->
+
+TcpClient <- : async_tcp(Task) runs
+    TcpClient -> Server: send request
+    TcpClient <-- Server
+TcpClient -->
+
+note over TcpClient, Server
+    Asynchronous
+end note
+
+TcpClient <- : async_tcp(Task) runs
```

Comment:
Don't show detail of TCP client, because its out of service context and has no additional value.